diff --git a/nbconvert/templates/html/basic.tpl b/nbconvert/templates/html/basic.tpl
index 9159415..e4f3606 100644
--- a/nbconvert/templates/html/basic.tpl
+++ b/nbconvert/templates/html/basic.tpl
@@ -23,11 +23,13 @@
 
 {% block in_prompt -%}
 <div class="prompt input_prompt">
-    {%- if cell.execution_count is defined -%}
-        In&nbsp;[{{ cell.execution_count|replace(None, "&nbsp;") }}]:
-    {%- else -%}
-        In&nbsp;[&nbsp;]:
-    {%- endif -%}
+    {% if not cell.metadata.hide_input %}  <!-- Exclude any cells that have the hide_input flag set (from the hide_input extension) -->
+        {%- if cell.execution_count is defined -%}
+            In&nbsp;[{{ cell.execution_count|replace(None, "&nbsp;") }}]:
+        {%- else -%}
+            In&nbsp;[&nbsp;]:
+        {%- endif -%}
+    {% endif %}
 </div>
 {%- endblock in_prompt %}
 
