--- /old/opt/conda/share/jupyter/nbextensions/validate_assignment/main.js
+++ /new/opt/conda/share/jupyter/nbextensions/validate_assignment/main.js
@@ -101,8 +101,10 @@ define([
 
             } else if (data.failed !== undefined) {
                 for (var i=0; i<data.failed.length; i++) {
-                    body.append($('<div/>').append($('<p/>').text('The following cell failed:')));
-                    body.append($('<pre/>').text(data.failed[i].source));
-                    body.append($('<pre/>').html(data.failed[i].error));
+                    if (!data.failed[i].hide_input) {
+                        body.append($('<div/>').append($('<p/>').text('The following cell failed:')));
+                        body.append($('<pre/>').text(data.failed[i].source));
+                    }
+                    body.append($('<pre class="failed_message"/>').html(data.failed[i].error));
                 }
                 body.addClass("validation-failed");
--- /old/opt/conda/share/jupyter/nbextensions/assignment_list/assignment_list.js
+++ /new/opt/conda/share/jupyter/nbextensions/assignment_list/assignment_list.js
@@ -564,9 +564,11 @@ define([
 
             } else if (data.failed !== undefined) {
                 for (var i=0; i<data.failed.length; i++) {
-                    body.append($('<div/>').append($('<p/>').text('The following cell failed:')));
-                    body.append($('<pre/>').text(data.failed[i].source));
-                    body.append($('<pre/>').html(data.failed[i].error));
+                    if (!data.failed[i].hide_input) {
+                        body.append($('<div/>').append($('<p/>').text('The following cell failed:')));
+                        body.append($('<pre/>').text(data.failed[i].source));
+                    }
+                    body.append($('<pre class="failed_message"/>').html(data.failed[i].error));
                 }
                 body.addClass("validation-failed");
                 this.validate_failure(button);
--- /old/opt/conda/lib/python3.7/site-packages/nbgrader/validator.py
+++ /new/opt/conda/lib/python3.7/site-packages/nbgrader/validator.py
@@ -252,8 +252,9 @@ class Validator(LoggingConfigurable):
                 results['failed'] = [{
                     "source": cell.source.strip(),
                     "error": ansi2html(self._extract_error(cell)),
-                    "raw_error": self._extract_error(cell)
-                } for cell in failed]
+                    "raw_error": self._extract_error(cell),
+                    "hide_input": cell.metadata.get("hide_input", False)
+                } for cell in failed]
 
         return results
 
