--- /old/opt/conda/lib/python3.7/site-packages/nbgrader/utils.py	2018-07-23 08:33:20.795242006 +0000
+++ /new/opt/conda/lib/python3.7/site-packages/nbgrader/utils.py	2018-07-23 08:33:04.679514671 +0000
@@ -152,6 +152,10 @@
 
 
 def get_username():
+    try:
+        return os.environ["JUPYTERHUB_USER"]
+    except:
+        pass
     """Get the username of the current process."""
     if pwd is None:
         raise OSError("get_username cannot be called on Windows")
