diff --git a/src/jupyter_contrib_nbextensions/nbextensions/freeze/main.js b/src/jupyter_contrib_nbextensions/nbextensions/freeze/main.js
index ff53618..4b575cb 100644
--- a/src/jupyter_contrib_nbextensions/nbextensions/freeze/main.js
+++ b/src/jupyter_contrib_nbextensions/nbextensions/freeze/main.js
@@ -160,23 +160,6 @@ define([
     }
 
     function load_extension () {
-        Jupyter.toolbar.add_buttons_group([
-            Jupyter.keyboard_manager.actions.register ({
-                help : 'lift restrictions from selected cells',
-                icon : 'fa-unlock-alt',
-                handler : make_normal_selected
-            }, 'make-cells-normal', mod_name),
-            Jupyter.keyboard_manager.actions.register({
-                help : 'make selected cells read-only',
-                icon: 'fa-lock',
-                handler : make_read_only_selected
-            }, 'make-cells-read-only', mod_name),
-            Jupyter.keyboard_manager.actions.register({
-                help : 'freeze selected cells',
-                icon : 'fa-asterisk',
-                handler : make_frozen_selected
-            }, 'freeze-cells', mod_name)
-        ]);
 
         patch_CodeCell_execute();
         patch_MarkdownCell_unrender();
diff --git a/src/jupyter_contrib_nbextensions/nbextensions/hide_input/main.js b/src/jupyter_contrib_nbextensions/nbextensions/hide_input/main.js
index c921c71..d6bc587 100644
--- a/src/jupyter_contrib_nbextensions/nbextensions/hide_input/main.js
+++ b/src/jupyter_contrib_nbextensions/nbextensions/hide_input/main.js
@@ -28,18 +28,6 @@ define([
     };
 
     var load_ipython_extension = function() {
-
-        // Add a button to the toolbar
-        $(Jupyter.toolbar.add_buttons_group([
-            Jupyter.keyboard_manager.actions.register({
-                help   : 'Toggle selected cell input display',
-                icon   : 'fa-chevron-up',
-                handler: function() {
-                    toggle_selected_input();
-                    setTimeout(function() { $('#btn-hide-input').blur(); }, 500);
-                }
-            }, 'toggle-cell-input-display', 'hide_input')
-        ])).find('.btn').attr('id', 'btn-hide-input');
         // Collapse all cells that are marked as hidden
         if (Jupyter.notebook !== undefined && Jupyter.notebook._fully_loaded) {
             // notebook already loaded. Update directly
